File: backend/src/main/java/com/project2/domain/member/entity/Member.java
package com.project2.domain.member.entity;

import java.util.ArrayList;
import java.util.Collection;
import java.util.List;

import org.springframework.security.core.GrantedAuthority;
import org.springframework.security.core.authority.SimpleGrantedAuthority;

import com.project2.domain.member.enums.Provider;
import com.project2.global.entity.BaseTime;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.experimental.SuperBuilder;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@SuperBuilder
@Entity
public class Member extends BaseTime {
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long id;

	@Column(nullable = false, unique = true)
	private String email;

	@Column(nullable = false, length = 50)
	private String nickname;

	private String profileImageUrl;

	@Enumerated(EnumType.STRING)
	@Column(columnDefinition = "VARCHAR(20)")
	private Provider provider;

	public Collection<? extends GrantedAuthority> getAuthorities() {
		return getMemberAuthoritiesAsString()
			.stream()
			.map(SimpleGrantedAuthority::new)
			.toList();
	}

	public List<String> getMemberAuthoritiesAsString() {

		return new ArrayList<>();
	}

	public String getProfileImageUrlOrDefaultUrl() {
		return (profileImageUrl == null || profileImageUrl.isBlank()) ? "https://placehold.co/640x640?text=O_O" : this.profileImageUrl;
	}
}

File: backend/src/main/java/com/project2/domain/member/entity/Member.kt
package com.project2.domain.member.entity

import com.project2.domain.member.enums.Provider
import com.project2.global.entity.BaseTime
import jakarta.persistence.*
import org.springframework.security.core.GrantedAuthority
import org.springframework.security.core.authority.SimpleGrantedAuthority
import java.time.LocalDateTime

@Entity
class Member() : BaseTime() {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null

    @Column(nullable = false, unique = true)
    var email: String = ""

    @Column(nullable = false, length = 50)
    var nickname: String = ""

    var profileImageUrl: String? = null

    @Enumerated(EnumType.STRING)
    @Column(columnDefinition = "VARCHAR(20)")
    var provider: Provider? = null

    companion object {
        @JvmStatic
        fun builder() = MemberBuilder()
    }

    class MemberBuilder {
        private var id: Long? = null
        private var email: String = ""
        private var nickname: String = ""
        private var profileImageUrl: String? = null
        private var provider: Provider? = null
        private var createdDate: LocalDateTime? = null

        fun id(id: Long?) = apply { this.id = id }
        fun email(email: String) = apply { this.email = email }
        fun nickname(nickname: String) = apply { this.nickname = nickname }
        fun profileImageUrl(profileImageUrl: String?) = apply { this.profileImageUrl = profileImageUrl }
        fun provider(provider: Provider) = apply { this.provider = provider }
        fun createdDate(createdDate: LocalDateTime) = apply { this.createdDate = createdDate }

        fun build() = Member().apply {
            this.id = this@MemberBuilder.id
            this.email = this@MemberBuilder.email
            this.nickname = this@MemberBuilder.nickname
            this.profileImageUrl = this@MemberBuilder.profileImageUrl
            this.provider = this@MemberBuilder.provider
            if (this@MemberBuilder.createdDate != null) {
                this.createdDate = this@MemberBuilder.createdDate!!
            }
        }
    }

    fun getAuthorities(): Collection<GrantedAuthority> =

            getMemberAuthoritiesAsString().map { SimpleGrantedAuthority(it) }

    fun getMemberAuthoritiesAsString(): List<String> =
            emptyList()

    fun getProfileImageUrlOrDefaultUrl(): String =
            profileImageUrl?.takeIf { it.isNotBlank() } ?: "/default-profile.png"

}